homeassistant/components/owlet/__init__.py

```diff
@@ -0,0 +1,95 @@
+"""The Owlet Smart Sock integration."""
+from __future__ import annotations
+
+import logging
+
+from pyowletapi.api import OwletAPI
+from pyowletapi.exceptions import (
+    OwletAuthenticationError,
+    OwletConnectionError,
+    OwletEmailError,
+    OwletPasswordError,
+)
+from pyowletapi.sock import Sock
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_API_TOKEN,
+    CONF_REGION,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_OWLET_EXPIRY, CONF_OWLET_REFRESH, DOMAIN, SUPPORTED_VERSIONS
+from .coordinator import OwletCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Owlet Smart Sock from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    owlet_api = OwletAPI(
+        region=entry.data[CONF_REGION],
+        token=entry.data[CONF_API_TOKEN],
+        expiry=entry.data[CONF_OWLET_EXPIRY],
+        refresh=entry.data[CONF_OWLET_REFRESH],
+        session=async_get_clientsession(hass),
+    )
+
+    try:
```

Comment:
Looks like you have things inside this try except that are not directly tied with the exceptions, you should move them outside the try except